src/pages/paymentServiceProviders/EditPaymentServiceProvider.tsx

```diff
@@ -356,6 +364,73 @@ export default class EditPaymentServiceProvider extends React.Component<IProps,
         this.setState({[section]: Object.assign({}, this.state.backup[section])} as any);
     }
 
+    debouncedChannelOptions = debounce((inputValue, callback) => {
+        this.promiseChannelOptions(inputValue, callback);
+    }, 500);
+
+
+    promiseChannelOptions(inputValue: string, callback: any) {
+        const limit = inputValue.length === 0 ? 10 : 99999;
+        const code = inputValue.length === 0 ? "" : inputValue;
+
+        this.context.instance.acquireTokenSilent({
+            ...loginRequest,
+            account: this.context.accounts[0]
+        })
+            .then((response: any) => {
+                apiClient.getChannels({
+                    Authorization: `Bearer ${response.idToken}`,
+                    ApiKey: "",
+                    page: 0,
+                    limit,
+                    code
+                }).then((resp: any) => {
+                    if (resp.right.status === 200) {
+                        const items: Array<any> = [];
+                        resp.right.value.channels.map((channel: any) => {
+                            // eslint-disable-next-line functional/immutable-data
+                            items.push({
+                                value: {channel_code: channel.channel_code, version: channel.version},
+                                label: channel.channel_code,
+                            });
+                        });
+                        callback(items);
+                    } else {
+                        callback([]);
+                    }
+                }).catch(() => {
+                    toast.error("Operazione non avvenuta a causa di un errore", {theme: "colored"});
+                    callback([]);
+                });
+            });
+    }
+
+
+    getPaymentTypesOptions(channel_code: string) {
```

Comment:
to avoid multiple requests to BE for the same channel (you make a request each time you open the select menù), you could store the output in a map and check in it if you have already get the info

Reply:
done 👍